driver/bpf/plumbing_helpers.h

```diff
@@ -423,93 +423,30 @@ static __always_inline bool drop_event(void *ctx,
 				       struct scap_bpf_settings *settings,
 				       enum syscall_flags drop_flags)
 {
-	if (!settings->dropping_mode)
+	if (drop_flags & UF_NEVER_DROP)
 		return false;
 
-	switch (evt_type) {
```

Comment:
Dropped `dropping` logic based on event type. Really useless considering that one can now completely disable syscalls through syscalls of interest mechanism.

Reply:
It's not useless. It's more fine-grained filtering than can be expressed via a syscall mask (e.g. dropping failed `bind()` exit events is fine but dropping successful ones is not: it modifies the fd in a way that sinsp would really want to know).

Having said that, I'm not sure it's valuable enough to keep as a special case.

Reply:
Yep; indeed i already wanted to deliver a patch to disable the capturing of failed syscalls; it could be a small trick to increase Falco perf and avoid filling the buffer with not-so-useful data, considering that a failed syscall can never be a security risk otherwise something bad is happening with your kernel.
Of course this would be opt in by exposing a sinsp API to enable this behavior.

Reply:
IMO let's keep this logic (in bpf and kmod) until _a patch to disable the capturing of failed syscalls_ lands. WDYT?

Reply:
np on my side!

Reply:
I removed last commit to keep the logic.